### PR TITLE
chore(security/TUI-286): update webpack, babel, storybook

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

we use storybook 3, webpack 3, and babel 6
all theses are very old and web pack has security issues.

**What is the chosen solution to this problem?**

update all theses one which are coupled.
Heavy use of `yarn list --pattern webpack` and others to track all subdependencies and align them. For example react-ace has been updated to align all dependencies with our deps.

# babel 7

first read https://babeljs.io/blog/2018/08/27/7.0.0
install babel-upgrade and execute it in every sub packages

ends with
```
	// babel 7
	"@babel/cli": "^7.0.0",
	"@babel/core": "^7.0.0",
	"@babel/plugin-proposal-class-properties": "^7.0.0",
	"@babel/plugin-proposal-object-rest-spread": "^7.0.0",
	"@babel/plugin-transform-object-assign": "^7.0.0",
	"@babel/plugin-proposal-export-namespace-from": "^7.0.0",
	"@babel/plugin-proposal-export-default-from": "^7.0.0",
	"@babel/preset-env": "^7.0.0",
	"@babel/preset-react": "^7.0.0",
```

Move .babelrc files to js files for those reasons :
* babel 7 babelrc does not support extends anymore. Replaced by `--root-mode upward`
* `--root-mode upward` is not compatible with the version of jest, that depends on babel 6.
remove all .babelrc file

Update the react.js builder in icons, was tricky.

# webpack 4

Update all versions
rework icons webpack config. following https://github.com/jeerbl/webfonts-loader documentation

# storybook 4

Following migration guide
https://medium.com/storybookjs/storybook-4-0-is-here-10b9857fc7de

# jest

jest is based on babel-core 6.

from the website

	Note: If you are using Babel version 7 then you need to install babel-jest, babel-core@^7.0.0-bridge.0 and @babel/core with the following command:
	`yarn add --dev babel-jest babel-core@^7.0.0-bridge.0 @babel/core regenerator-runtime`
	You will need to use babel.config.js in order to transpile node_modules

# i18next

We are very late on i18next but I will update it in an other PR. I had many issues while update all the stack with the `components/src/translate.test.js` file.

I have added to test-setup.js a mock of i18next lib which force me to update two tests but it decrease so much the sizes of the snapshots !

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

Warning: You should not need to migrate anything in your project after this PR merged.
But If you want to use webpack 4 I recommand you to look at this PR (description and diff) it will help you to do so in your project.

you can tests the PR using the pre-release `1.10.0-webpack4`

### CODE REVIEW

File: test-setup.js [hunk 1]
```diff
@@ -1,10 +1,64 @@
...
+				buff = split.reduce((acc, current) => {
+					const sub = current.split('{{');
+					let value = sub.length > 1 ? options[sub[1]] : '';
+					if (value === undefined) {
+						value = '';
+					}
+					return `${acc}${sub[0]}${value}`;
+				}, '');
+			}
+			return buff;
```
Comment (jmfrancois) [inline test-setup.js]: I would love to extract out this function but jest.mock do not let me doing so, the code is hoisted by jest.

File: versions/version.js [hunk 2]
```diff
@@ -55,17 +55,24 @@ const VERSIONS = require('./dependencies');
...
 
 const files = [
+	path.join(__dirname, '../npm-audit/package.json'),
 	path.join(__dirname, '../packages/cmf/package.json'),
 	path.join(__dirname, '../packages/cmf-cqrs/package.json'),
 	path.join(__dirname, '../packages/cmf-webpack-plugin/package.json'),
 	path.join(__dirname, '../packages/components/package.json'),
 	path.join(__dirname, '../packages/containers/package.json'),
+	path.join(__dirname, '../packages/datagrid/package.json'),
+	path.join(__dirname, '../packages/eslint-config/package.json'),
```
Comment (jsomsanith) [inline versions/version.js]: Discussed together, can you remove this one for now ?

File: versions/version.js [hunk 3]
```diff
@@ -55,17 +55,24 @@ const VERSIONS = require('./dependencies');
...
 	path.join(__dirname, '../packages/forms/package.json'),
 	path.join(__dirname, '../packages/generator/package.json'),
+	path.join(__dirname, '../packages/generator/generators/app/templates/package.json'),
+	path.join(__dirname, '../packages/html-webpack-plugin/package.json'),
 	path.join(__dirname, '../packages/icons/package.json'),
+	// path.join(__dirname, '../packages/logging/package.json'),
```
Comment (jsomsanith) [inline versions/version.js]: Better remove than comment

File: packages/storybook-cmf/src/index.js [hunk 4]
```diff
@@ -1,37 +1,24 @@
...
-				>
-					{storyFn()}
-				</CMFStory>
-			),
-			{ showInline: true },
-		);
-	},
-};
+export default makeDecorator({
+	name: 'withNotes',
```
[the remark this answers is not in the document]
Comment (jmfrancois) [reply packages/storybook-cmf/src/index.js]: withCMF is better

**Pull request merged**